### PR TITLE
Go-go go Modules

### PR DESCRIPTION
**Pull request by justnoise**
In the work to test out virtual kubelet changes, I ran into some k8s dependency issues.  Dep wasn't going very well so I moved the repo over to modules, something we were interested in doing anyways.

Etcd has a few issues around how they handle their dependencies. As such, we had to downgrade etcd from 3.3.18 to 3.3.13 and make some tweaks to `go.mod`.  I tested the backwards/forwards compatibility and it seems to be a-ok (which it should be).

The changes to the go-client mostly impact the cell controller.  For now I've added an issue to timeout our calls to k8s and left the contexts as TODOs. That's functionally the same as a background context but we should updated them to timeouts.

In the build, when making the binaries go will rewrite go.mod and go.sum despite the fact that those have been tidied.  I tried building with `-mod=readonly` but that did not work.  To keep our binaries from being marked with dirty tags, I added a `git checkout -- go.mod go.sum` to the build instructions after the make step.  There might be a better way around this.

Also got rid of the vendor folder.  The sweet sweet code deletions are MINE!

### CODE REVIEW

**Comment by justnoise** [comment]
One question: do we want to keep `vendor/`?

**Review by ldx — APPROVED** [review]
I went ahead and added a commit that ensure that installing deepcopy-gen won't update go.mod as an unwanted side effect. Other than that the only thing missing is making the go build targets depend on go.mod and go.sum.

File: .travis.yml [hunk 1]
```diff
@@ -5,6 +5,10 @@ sudo: required
 dist: xenial
 services:
 - docker
+cache:
+  directories:
+    - $HOME/.cache/go-build
+    - $HOME/gopath/pkg/mod
```
[the remark this answers is not in the document]
Comment (justnoise) [reply .travis.yml:11]: Ran a simple test without it and with it:
With the cache: Setting up the cache takes ~22 seconds, `make` takes 13 seconds
Without the cache: Setting up the cache takes 0 seconds, make takes 151 seconds

So the cache appears to save us almost 2 minutes.

File: .travis.yml [hunk 2]
```diff
@@ -38,6 +42,7 @@ before_script:
 script:
 - set -e
 - make
+- 'git checkout -- go.mod go.sum'
```
[the remark this answers is not in the document]
Comment (justnoise) [reply .travis.yml]: I think you found it!

**Comment by justnoise** [comment]
@ldx one last thing for this one: I'm going to add the `vendor/` directory to `.gitignore`.

**Comment by justnoise** [comment]
retest this please

**Pull request merged**